Backpack/src/main/java/net/skyscanner/backpack/text/BpkText.kt

```diff
@@ -69,23 +91,29 @@ open class BpkText(
       defStyleAttr, 0)
 
     textStyle = a.getInt(R.styleable.BpkText_textStyle, BASE)
-    emphasize = a.getBoolean(R.styleable.BpkText_emphasize, false)
+    val weightArg = a.getInt(R.styleable.BpkText_weight, -1)
+    if (weightArg == -1) {
+      // if weight has not been set we still read the emphasize property
+      emphasize = a.getBoolean(R.styleable.BpkText_emphasize, false)
+    } else {
+      weight = Weight.values()[weightArg]
+    }
 
     a.recycle()
   }
 
   private fun setup() {
 
     val styleProps = styleMapping[textStyle]
+    styleProps ?: throw IllegalStateException("Invalid textStyle")
+
+    val textAppearance = styleProps[weight.ordinal]
+    textAppearance ?: throw IllegalStateException("Weight $weight is not supported for the current size")
 
     if (textStyle == CAPS) {
       isAllCaps = true
     }
 
-    if (styleProps != null) {
-      TextViewCompat.setTextAppearance(this, if (emphasize) styleProps.first else styleProps.second)
```

Comment:
Without this line, how does `emphasize` get applied? We still want it to take effect right, if `weight` isn't passed in. I might have just missed it.

Reply:
Is still being set on line 117, I just moved the null check logic up. `TextViewCompat.setTextAppearance(this, textAppearance)` this is what applies all the styles to the textview.

Reply:
Ah yeah - nice 😄